tests/cxx/subfolder/direct_subfolder.h

```diff
@@ -0,0 +1,19 @@
+//===--- direct_subfolder.h - test input file for iwyu --------------------===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+// This file includes and uses only another file in the subfolder
+
+#ifndef DEVTOOLS_MAINTENANCE_INCLUDE_WHAT_YOU_USE_TESTS_DIRECT_SUBFOLDER_H_
+#define DEVTOOLS_MAINTENANCE_INCLUDE_WHAT_YOU_USE_TESTS_DIRECT_SUBFOLDER_H_
+
+#include "indirect_subfolder.h"
+
+namespace { IndirectSubfolderClass foo; }
```

Comment:
OK, I think I can follow your reasoning, but I wonder if it's relevant to the indirect subfolder case. IWYU doesn't care about uses in files it doesn't analyze (i.e. that are not associated to the main source file or listed in `--check_also`), so this is really just adding noise.

If you disagree, can you clarify the test to reflect your reasoning?

An anonymous namespace in a header file is suspect, so maybe you can use something like this instead:

    void f() {
      // Use the class in this header to see that IWYU
      // sees through the include chain and recommends including
      // indirect_subfolder.h even if this header satisfies the dependency
      // on IndirectSubfolderClass.
      IndirectSubfolderClass x;
    }

Comment is just an example, please write something there that reflects your intent.